doc/coq.rst

```diff
@@ -0,0 +1,494 @@
+.. _coq:
+
+***
+Coq
+***
+
+
+Dune is able to build Coq projects and plugins with support for source file
+preprocessing, extraction and documenation.
+
+
+.. contents:: Table of Contents
+    :depth: 3
+
+.. _coq-theory:
+
+coq.theory
+----------
+
+A Coq project consists of Coq ``.v`` files and optional OCaml libraries linking
+to the Coq API (in which case we say the project is a *Coq plugin*). To enable
+Coq support in a Dune project, the language version should be selected in the
+``dune-project`` file. For example:
+
+.. code:: scheme
+
+    (using coq 0.3)
+
+This will enable support for the ``coq.theory`` stanza in the current project.
+If the language version is absent, Dune will automatically add this line with
+the latest Coq version to the project file once a ``(coq.theory ...)`` stanza is
+used anywhere.
+
+The basic form for defining Coq libraries is very similar to the OCaml form:
+
+.. code:: scheme
+
+    (coq.theory
+     (name <module_prefix>)
+     (package <package>)
+     (synopsis <text>)
+     (modules <ordered_set_lang>)
+     (libraries <ocaml_libraries>)
+     (flags <coq_flags>)
+     (mode <coq_native_mode>)
+     (theories <coq_theories>))
+
+The stanza will build all ``.v`` files on the given directory. The semantics of
+fields are:
+
+- ``<module_prefix>`` is a dot-separated list of valid Coq module names and
+  determines the module scope under which the theory is compiled [``-R``
+  option]. For example, if ``<module_prefix>`` is ``foo.Bar``, the theory
```

Comment:
```suggestion
- ``<module_prefix>`` is the module prefix for the Coq package (equivalent to the
  ``-R`` command line option passed to Coq).
  For example, if ``<module_prefix>`` is ``foo.Bar``, the theory
```
This terminology isn't very consistent with the terms we use in the Coq documentation.  It doesn't use "theory" in this sense at all.  The Coq documentation calls the result of compiling a project a package (as does OPAM).

BTW, shouldn't there be a way to specify -Q?

Reply:
I don't know if we define "theory" in the Coq documentation actually. Perhaps at the beginning I should mention that a collection of .v files is called a theory.

With regards to specifying -Q, the default is -Q for theories you depend on and -R for your own. By design this should nto be configurable. See https://github.com/ocaml/dune/issues/2585

Reply:
I've not seen "theory" used this way in the Coq doc, and I've been looking at it quite a bit recently, so best not to introduce a new term.

Reply:
I am not the one to introduce this term, but I have given it a technically precise definition in the begining. @ejgallego what do you think?